src/FC/GPS.cpp

```diff
@@ -0,0 +1,84 @@
+#include "GPS.h"
+#include <Wire.h>
+
+
+namespace GPS {
+
+    SFE_UBLOX_GNSS myGNSS;
+    uint32_t gpsUpdatePeriod = 40 * 1000;
+
+    Comms::Packet latLongPacket = {.id = 11};
+    double latitude;
+    double longitude;
+
+
+    Comms::Packet auxPacket = {.id = 12};
+    double altitude;
+    double speed;
+    double angle;
+    byte SIV;
+
+    //
+    void initGPS() {
+        Serial.begin(9600);
+        Wire.begin();
+        Serial.println("GPS setup");
+        if (!myGNSS.begin()) {
+            Serial.println("GPS failed to start");
+        } else {
+            Serial.println("GPS started");
+        }
+        // Version 2 - UBX
+        myGNSS.setI2COutput(COM_TYPE_UBX);   //Set the I2C port to output UBX only (turn off NMEA noise)
+        myGNSS.saveConfigSelective(VAL_CFG_SUBSEC_IOPORT); //Save (only) the communications port settings to flash and BBR
```

Comment:
On the flight computer, the GPS is connected to the teensy using SPI, so can you set up the SPI interface here instead of I2C? Also, don't use Serial.print for logging. If you want to log messages for debugging purposes, use the `DEBUG("<message>");` macro